### DISCUSSION TITLE
Query EntityStore from EntityUIQuery

### DISCUSSION THREAD

**Discussion by lilas78** [salesforce/akita#575] [Q&A]
## I'm submitting a ...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[x ] Support request
[ ] Other... Please describe:
</code></pre>

## Question

I have an entity state with a entity UI state 

export interface ItemState extends EntityState<Item> {}
export type ItemUI = {
    isOpen: boolean;
    isLoading: boolean;
}
export interface ItemUIState extends EntityState<ItemUI> { }

I created the UIStore and the UIQuery as requested in the documentation
I want to query for all items that have the UIState property 'isOpen' to true
This query 

        openedItemsUI$ = store.ui.selectAll({
            filterBy: state => state.isOpen === true,
        })
		
returns an observable on an array of ItemUI. I can see with devtools that this object has an "id" field which is the same as the id of the entity it refers to. Bu this id is not accessible in code.

How can I get an observable on the array of Item corresponding to the selected ItemUI ?	

</code></pre>

**Comment by NetanelBasal** [comment]
See the last section - https://engineering.datorama.com/ui-state-management-made-easy-with-akita-and-angular-6e460566ec7c

**Comment by lilas78** [comment]
Thank you for your quick anwser.
I saw this code, but this doesn't work as I expect it.

Here is my code adapted with your snippet :

@Injectable({ providedIn: 'root' })
export class DashboardItemQuery extends QueryEntity<DashboardItemState> {
    ui: EntityUIQuery<DashboardItemUIState>;
    constructor(protected store: DashboardItemStore) {
        super(store);
        this.createUIQuery();
    }

    selectEntityWithUI(id: ID): Observable<(DashboardItem & DashboardItemUI)[]> {
        const widgets = this.selectAll();
        const widgetsUI = this.ui.select(
            state => { state.isOpen === true }
        );

        return combineLatest(
            widgets,
            widgetsUI
        ).pipe(map(([widgets, widgetsUI]) => {
            return widgets.map(widget => {
                return {
                    ...widget,
                    ...widgetsUI[widget.id]
                }
            })
        }))
    }
}

But my observable is not updated when a ui state property isOpen is updated.

**Comment by lilas78** [comment]
OK, I get notified now. But the result is not what I expect. I want to filter the array of DashboardItem, and to keep those  which have their UI state isOpen equal to true

Here I get all the DashboardItems

**Comment by lilas78** [comment]
I would like to do something like that :

```
 selectOpenedEntity(): Observable<DashboardItem[]> {
        const widgets = this.selectAll();
        const widgetsUI = this.ui.selectAll({
            filterBy: state => state.isOpen === true,
        })

        return combineLatest([
            widgets,
            widgetsUI]
        ).pipe(map(([widgets, widgetsUI]) => {
            return widgets.filter(widget => {
              //TODO 
              //Returns if widgetsUI contains an object with id widget.id
              // but can't access DashboardItemUI.id
            })
	      }))

	  }
```
